### ISSUE TITLE
[Bug] No music in Linux

### ISSUE THREAD

**Issue by weirddan455** [CorsixTH/CorsixTH#1815]
#### Describe the issue
I'm not getting any background music in Linux.  I have both Timidity++ and FluidSynth installed (and either works fine in DosBox.)  I was getting the following error in my gamelog so I created the cfg file it was looking for:

`/usr/share/corsix-th/Lua/audio.lua:577: Could not load music file 'Sound/Midi/STEADY.XMI' (Couldn't open /etc/timidity++/freepats.cfg)`

I just put the following in there (which is the same as is in my timidity.cfg)

`soundfont /usr/share/soundfonts/freepats-general-midi.sf2`

Afterwards, it's stopped printing that file not found error but there's still no sound (and no more relevant errors.)

Does this game work like DosBox where it expects the daemon to be running and it sends sound out the MIDI port or what?  I'm not seeing any configuration options for MIDI sound in the game.  I know some other games (like gzDoom) just use FluidSynth as a library and don't require the daemon.

#### Steps to Reproduce
1. Load game

2. No music :(

#### Expected Behavior
Music should be playing

#### System Information
**CorsixTH Version:** 0.64.r309.g361b7faf-1

**Operating System:** Arch Linux

**Theme Hospital Version:** Demo

#### Gamelog.txt
```
Created game with savegame version 152.
Error in callback handler: 
/usr/share/corsix-th/Lua/audio.lua:577: Could not load music file 'Sound/Midi/STEADY.XMI' (Couldn't open /etc/timidity++/freepats.cfg)
stack traceback:
	[C]: in function 'coroutine.yield'
	/usr/share/corsix-th/Lua/app.lua:1028: in function </usr/share/corsix-th/Lua/app.lua:1023>

```

**Comment by TheCycoONE** [comment]
I don't think this is our bug, but I agree the situation on ArchLinux is unfortunate. See my comment at: https://aur.archlinux.org/packages/corsix-th#comment-695085

The environment variable is important, without it sdl2_mixer tries to use timidity++ and the timidity++ as packaged by archlinux isn't compatible.

**Comment by kracas** [comment]
Tried everything to get the midi working on Arch, finally gave up and got a [snap](https://snapcraft.io/corsixth). I know it's probably not the best solution, but at least it works.

**Comment by Alberth289346** [comment]
Can we check for this, and warn the user?

**Comment by TheCycoONE** [comment]
The missing config file maybe, it's clearly reported to us by sdl2_mixer. The yet to be released 2.0.5 likely fixes the issue (or at least puts more control over the timidity configuration on the distro. The incompatible soundfont or whatever was causing the silent error - I don't think so.

For what it's worth on Arch I have midi by using fluidsynth instead of timidity: https://aur.archlinux.org/packages/corsix-th/#comment-695085
On the same AUR there are reports of people having success with freepats-legacy

**Comment by Alberth289346** [comment]
I was thinking to test for the environment variable, and give a warning if missing, possibly with a more extensive description in the readme file.

If the contents of the variable is fixed, you could even create it, I'd guess before initializing the SDL library. It does have the problem of being very distro specific though.

**Comment by ell-be** [comment]
Using CorsixTH v0.65.1 on Arch Linux - initially - I ran into the same problem.
Now music is playing with timidity++ 2.15.0-5 in combination with freepats-legacy 20060219-1:

-  https://aur.archlinux.org/packages/freepats-legacy,
-  copying /usr/share/freepats/timidity-freepats.cfg to /etc/timidity++/timidity.cfg.

**Comment by knightprog** [comment]
I can confirm the music works fine on ArchLinux with fluidsynth, by setting the environment variable SDL_SOUNDFONTS.  
You can try to launch game like this in command line:
```
SDL_SOUNDFONTS=/usr/share/soundfonts/FluidR3_GM.sf2 corsix-th
```

If it works, you can launch it via the launcher icon, just copy the file `/usr/share/applications/com.corsixth.corsixth.desktop` in  `~/.local/share/applications/com.corsixth.corsixth.desktop"`.  

Then open the file  `~/.local/share/applications/com.corsixth.corsixth.desktop"` and edit the `Exec` line like this:
``` 
Exec=bash -c "SDL_SOUNDFONTS=/usr/share/soundfonts/FluidR3_GM.sf2 corsix-th"
```

**CorsixTH Version:** Version: 0.66_release-1
**Fluidsynth Version:** 2.3.2-1

**Comment by lewri** [comment]
Should now be fixed by #2322  -- midi playback will be fully done inside CorsixTH moving forward.
@TheCycoONE can you confirm before closing?

**Comment by TheCycoONE** [comment]
Fixed as long as fluidsynth is installed yes.
